app/src/main/java/org/oppia/android/app/onboarding/OnboardingFragmentPresenter.kt

```diff
@@ -45,9 +48,15 @@ class OnboardingFragmentPresenter @Inject constructor(
   }
 
   private fun setUpViewPager() {
-    val adapter =
-      OnboardingPagerAdapter(fragment.requireContext(), getOnboardingSlideFinalViewModel())
-    binding.onboardingSlideViewPager.adapter = adapter
+    binding.onboardingSlideViewPager.adapter = createRecyclerViewAdapter()
+    binding.onboardingSlideViewPager.adapter.setData(
+      listOf(
+        OnboardingSlideViewModel(context = activity, viewPagerSlide = ViewPagerSlide.SLIDE_0),
+        OnboardingSlideViewModel(context = activity, viewPagerSlide = ViewPagerSlide.SLIDE_1),
+        OnboardingSlideViewModel(context = activity, viewPagerSlide = ViewPagerSlide.SLIDE_2),
+        getOnboardingSlideFinalViewModel()
+      )
+    )
```

Comment:
It's showing this compile time error - 
```
Smart cast to '({Adapter<(androidx.recyclerview.widget.RecyclerView.ViewHolder..androidx.recyclerview.widget.RecyclerView.ViewHolder?)> & BindableAdapter<OnboardingViewPagerViewModel>}..BindableAdapter<OnboardingViewPagerViewModel>)' is impossible, because 'binding.onboardingSlideViewPager.adapter' is a complex expression
```

Reply:
Ah, something weird is happening here. Can you try saving the adapter on the line above to a separate variable, assigning the view pager's adapter to that, then calling setData() directly on the adapter?

Reply:
It worked :)